crux-agent/demo_continuation_fallback.py

```
@@ -0,0 +1,271 @@
+• When continuation is requested with evolution_history and current iteration fails to generate valid output
+• Instead of raising a hard exception, it falls back to the latest valid iteration
+• Marks should_stop = True and metadata.fallback_used = True with diagnostic message
+• The allow_continuation_fallback flag (default True) can be used to toggle this behavior
+
+Run with: python demo_continuation_fallback.py
+"""
+
+import asyncio
+from unittest.mock import MagicMock
+
+from app.core.engine.self_evolve import Problem, SelfEvolve, Solution
```

Ruff (0.12.2)

`app.core.engine.self_evolve.Solution` imported but unused

Remove unused import: `app.core.engine.self_evolve.Solution`

(F401)